src/topology/stone_cech.lean

```diff
@@ -307,4 +308,53 @@ end
 instance stone_cech.compact_space : compact_space (stone_cech α) :=
 quotient.compact_space
 
+section universal_property
+
+variables {δ : Type u} [topological_space δ]
+variables {δ₁ : Type u} [topological_space δ₁] [t2_space δ₁] [compact_space δ₁]
+variables {δ₂ : Type u} [topological_space δ₂] [t2_space δ₂] [compact_space δ₂]
+
+/-- The universal property that uniquely characterizes the Stone-Čech compactification of `α`, when
+  `stone_cech_unit` is an embedding. Notice that `stone_cech α` has this property
+  in general by `stone_cech_extend_extends`, `continuous_stone_cech_extend` and
+  `stone_cech_hom_ext`, but lemma `stone_cech_extension_property_unique` assumes a dense
+  embedding that `stone_cech_unit` is not guaranteed to be. -/
+def stone_cech_extension_property (e : α → δ) :=
+  ∀ {γ : Type u} [topological_space γ], by exactI ∀ [t2_space γ]
+  [compact_space γ] (f : α → γ) (hf: continuous f),
+  ∃! (ext_f : δ → γ), continuous ext_f ∧ f = ext_f ∘ e
+
+/-- Up to isomorphism, there can only be one space `α` is densely embedded in, that satisfies
+  `stone_cech_extension_property`. -/
+def stone_cech_extension_property_unique
+  {e₁ : α → δ₁} {e₂ : α → δ₂} (emb₁ : dense_embedding e₁) (emb₂ : dense_embedding e₂)
+  (he₁ : stone_cech_extension_property e₁) (he₂ : stone_cech_extension_property e₂) : δ₁ ≃ₜ δ₂ :=
+begin
+  obtain ⟨c₁, d₁, c₂, d₂⟩ : continuous e₁ ∧ dense_range e₁ ∧ continuous e₂ ∧ dense_range e₂,
```

Comment:
I'm surprised by how long this proof is. I think you can deduplicate `left_inv` and `right_inv` using a helper lemma.